modules/failure-handler/src/main/java/org/apache/ignite/internal/failure/FailureManager.java

```diff
@@ -160,14 +158,14 @@ private synchronized boolean process(FailureContext failureCtx, FailureHandler h
             return false;
         }
 
-        var exceptionForLogging = new StackTraceCapturingException(failureCtx.error());
+        var exceptionForLogging = new StackTraceCapturingException(failureCtx.message(), failureCtx.error());
         if (handler.ignoredFailureTypes().contains(failureCtx.type())) {
             LOG.warn(IGNORED_FAILURE_LOG_MSG, exceptionForLogging, handler, failureCtx.type());
         } else {
             LOG.error(FAILURE_LOG_MSG, exceptionForLogging, handler, failureCtx.type());
         }
 
-        if (reserveBuf != null && hasCauseOrSuppressed(failureCtx.error(), OutOfMemoryError.class)) {
+        if (reserveBuf != null && failureCtx.error() != null && hasCauseOrSuppressed(failureCtx.error(), OutOfMemoryError.class)) {
```

Comment:
What if `hasCauseOrSuppressed` would just return `false` if we pass null into it?

Reply:
Alright, that's fair